File: Examples/Chart/WebForm/VB.NET/App_Code/DataManipulation/Binding/dsTable.vb
Imports Microsoft.VisualBasic
	Imports System
	Imports System.Data
	Imports System.Xml
	Imports System.Runtime.Serialization
Namespace Nevron.Examples.Chart.WebForm

	<Serializable(), System.ComponentModel.DesignerCategoryAttribute("code"), System.Diagnostics.DebuggerStepThrough(), System.ComponentModel.ToolboxItem(True)> _
	Public Class dsTable
		Inherits DataSet

		Private tableMyTable As MyTableDataTable

		Public Sub New()
			Me.InitClass()
			Dim schemaChangedHandler As System.ComponentModel.CollectionChangeEventHandler = New System.ComponentModel.CollectionChangeEventHandler(AddressOf Me.SchemaChanged)
			AddHandler Me.Tables.CollectionChanged, schemaChangedHandler
			AddHandler Me.Relations.CollectionChanged, schemaChangedHandler
		End Sub

		Protected Sub New(ByVal info As SerializationInfo, ByVal context As StreamingContext)
			Dim strSchema As String = (CStr(info.GetValue("XmlSchema", GetType(String))))
			If (Not strSchema Is Nothing) Then
				Dim ds As DataSet = New DataSet()
				ds.ReadXmlSchema(New XmlTextReader(New System.IO.StringReader(strSchema)))
				If (Not ds.Tables("MyTable") Is Nothing) Then
					Me.Tables.Add(New MyTableDataTable(ds.Tables("MyTable")))
				End If
				Me.DataSetName = ds.DataSetName
				Me.Prefix = ds.Prefix
				Me.Namespace = ds.Namespace
				Me.Locale = ds.Locale
				Me.CaseSensitive = ds.CaseSensitive
				Me.EnforceConstraints = ds.EnforceConstraints
				Me.Merge(ds, False, System.Data.MissingSchemaAction.Add)
				Me.InitVars()
			Else
				Me.InitClass()
			End If
			Me.GetSerializationData(info, context)
			Dim schemaChangedHandler As System.ComponentModel.CollectionChangeEventHandler = New System.ComponentModel.CollectionChangeEventHandler(AddressOf Me.SchemaChanged)
			AddHandler Me.Tables.CollectionChanged, schemaChangedHandler
			AddHandler Me.Relations.CollectionChanged, schemaChangedHandler
		End Sub

		<System.ComponentModel.Browsable(False), System.ComponentModel.DesignerSerializationVisibilityAttribute(System.ComponentModel.DesignerSerializationVisibility.Content)> _
		Public ReadOnly Property MyTable() As MyTableDataTable
			Get
				Return Me.tableMyTable
			End Get
		End Property

		Public Overrides Function Clone() As DataSet
			Dim cln As dsTable = (CType(MyBase.Clone(), dsTable))
			cln.InitVars()
			Return cln
		End Function

		Protected Overrides Function ShouldSerializeTables() As Boolean
			Return False
		End Function

		Protected Overrides Function ShouldSerializeRelations() As Boolean
			Return False
		End Function

		Protected Overrides Sub ReadXmlSerializable(ByVal reader As XmlReader)
			Me.Reset()
			Dim ds As DataSet = New DataSet()
			ds.ReadXml(reader)
			If (Not ds.Tables("MyTable") Is Nothing) Then
				Me.Tables.Add(New MyTableDataTable(ds.Tables("MyTable")))
			End If
			Me.DataSetName = ds.DataSetName
			Me.Prefix = ds.Prefix
			Me.Namespace = ds.Namespace
			Me.Locale = ds.Locale
			Me.CaseSensitive = ds.CaseSensitive
			Me.EnforceConstraints = ds.EnforceConstraints
			Me.Merge(ds, False, System.Data.MissingSchemaAction.Add)
			Me.InitVars()
		End Sub

		Protected Overrides Function GetSchemaSerializable() As System.Xml.Schema.XmlSchema
			Dim stream As System.IO.MemoryStream = New System.IO.MemoryStream()
			Me.WriteXmlSchema(New XmlTextWriter(stream, Nothing))
			stream.Position = 0
			Return System.Xml.Schema.XmlSchema.Read(New XmlTextReader(stream), Nothing)
		End Function

		Friend Sub InitVars()
			Me.tableMyTable = (CType(Me.Tables("MyTable"), MyTableDataTable))
			If (Not Me.tableMyTable Is Nothing) Then
				Me.tableMyTable.InitVars()
			End If
		End Sub

		Private Sub InitClass()
			Me.DataSetName = "dsTable"
			Me.Prefix = ""
			Me.Namespace = "http://www.tempuri.org/dsTable.xsd"
			Me.Locale = New System.Globalization.CultureInfo("bg-BG")
			Me.CaseSensitive = False
			Me.EnforceConstraints = True
			Me.tableMyTable = New MyTableDataTable()
			Me.Tables.Add(Me.tableMyTable)
		End Sub

		Private Function ShouldSerializeMyTable() As Boolean
			Return False
		End Function

		Private Sub SchemaChanged(ByVal sender As Object, ByVal e As System.ComponentModel.CollectionChangeEventArgs)
			If (e.Action = System.ComponentModel.CollectionChangeAction.Remove) Then
				Me.InitVars()
			End If
		End Sub

		Public Delegate Sub MyTableRowChangeEventHandler(ByVal sender As Object, ByVal e As MyTableRowChangeEvent)

		<System.Diagnostics.DebuggerStepThrough()> _
		Public Class MyTableDataTable
			Inherits DataTable
			Implements System.Collections.IEnumerable

			Public columnid As DataColumn
			Public columncolors As DataColumn
			Public columnvalues As DataColumn

			Friend Sub New()
				MyBase.New("MyTable")
				Me.InitClass()
			End Sub

			Friend Sub New(ByVal table As DataTable)
				MyBase.New(table.TableName)
				If (table.CaseSensitive <> table.DataSet.CaseSensitive) Then
					Me.CaseSensitive = table.CaseSensitive
				End If
				If (table.Locale.ToString() <> table.DataSet.Locale.ToString()) Then
					Me.Locale = table.Locale
				End If
				If (table.Namespace <> table.DataSet.Namespace) Then
					Me.Namespace = table.Namespace
				End If
				Me.Prefix = table.Prefix
				Me.MinimumCapacity = table.MinimumCapacity
				Me.DisplayExpression = table.DisplayExpression
			End Sub

			<System.ComponentModel.Browsable(False)> _
			Public ReadOnly Property Count() As Integer
				Get
					Return Me.Rows.Count
				End Get
			End Property

			Friend ReadOnly Property idColumn() As DataColumn
				Get
					Return Me.columnid
				End Get
			End Property

			Friend ReadOnly Property colorsColumn() As DataColumn
				Get
					Return Me.columncolors
				End Get
			End Property

			Friend ReadOnly Property valuesColumn() As DataColumn
				Get
					Return Me.columnvalues
				End Get
			End Property

			Public ReadOnly Default Property Item(ByVal index As Integer) As MyTableRow
				Get
					Return (CType(Me.Rows(index), MyTableRow))
				End Get
			End Property

			Public Event MyTableRowChanged As MyTableRowChangeEventHandler

			Public Event MyTableRowChanging As MyTableRowChangeEventHandler

			Public Event MyTableRowDeleted As MyTableRowChangeEventHandler

			Public Event MyTableRowDeleting As MyTableRowChangeEventHandler

			Public Sub AddMyTableRow(ByVal row As MyTableRow)
				Me.Rows.Add(row)
			End Sub

			Public Function AddMyTableRow(ByVal colors As String, ByVal values As Integer) As MyTableRow
				Dim rowMyTableRow As MyTableRow = (CType(Me.NewRow(), MyTableRow))
				rowMyTableRow.ItemArray = New Object() { Nothing, colors, values}
				Me.Rows.Add(rowMyTableRow)
				Return rowMyTableRow
			End Function

			Public Function FindByid(ByVal id As Integer) As MyTableRow
				Return (CType(Me.Rows.Find(New Object() { id}), MyTableRow))
			End Function

			Public Function GetEnumerator() As System.Collections.IEnumerator Implements System.Collections.IEnumerable.GetEnumerator
				Return Me.Rows.GetEnumerator()
			End Function

			Public Overrides Function Clone() As DataTable
				Dim cln As MyTableDataTable = (CType(MyBase.Clone(), MyTableDataTable))
				cln.InitVars()
				Return cln
			End Function

			Protected Overrides Function CreateInstance() As DataTable
				Return New MyTableDataTable()
			End Function

			Friend Sub InitVars()
				Me.columnid = Me.Columns("id")
				Me.columncolors = Me.Columns("colors")
				Me.columnvalues = Me.Columns("values")
			End Sub

			Private Sub InitClass()
				Me.columnid = New DataColumn("id", GetType(Integer), Nothing, System.Data.MappingType.Element)
				Me.Columns.Add(Me.columnid)
				Me.columncolors = New DataColumn("colors", GetType(String), Nothing, System.Data.MappingType.Element)
				Me.Columns.Add(Me.columncolors)
				Me.columnvalues = New DataColumn("values", GetType(Integer), Nothing, System.Data.MappingType.Element)
				Me.Columns.Add(Me.columnvalues)
				Me.Constraints.Add(New UniqueConstraint("Constraint1", New DataColumn() { Me.columnid}, True))
				Me.columnid.AutoIncrement = True
				Me.columnid.AllowDBNull = False
				Me.columnid.Unique = True
			End Sub

			Public Function NewMyTableRow() As MyTableRow
				Return (CType(Me.NewRow(), MyTableRow))
			End Function

			Protected Overrides Function NewRowFromBuilder(ByVal builder As DataRowBuilder) As DataRow
				Return New MyTableRow(builder)
			End Function

			Protected Overrides Function GetRowType() As System.Type
				Return GetType(MyTableRow)
			End Function

			Protected Overrides Sub OnRowChanged(ByVal e As DataRowChangeEventArgs)
				MyBase.OnRowChanged(e)
				If (Not Me.MyTableRowChangedEvent Is Nothing) Then
					RaiseEvent MyTableRowChanged(Me, New MyTableRowChangeEvent((CType(e.Row, MyTableRow)), e.Action))
				End If
			End Sub

			Protected Overrides Sub OnRowChanging(ByVal e As DataRowChangeEventArgs)
				MyBase.OnRowChanging(e)
				If (Not Me.MyTableRowChangingEvent Is Nothing) Then
					RaiseEvent MyTableRowChanging(Me, New MyTableRowChangeEvent((CType(e.Row, MyTableRow)), e.Action))
				End If
			End Sub

			Protected Overrides Sub OnRowDeleted(ByVal e As DataRowChangeEventArgs)
				MyBase.OnRowDeleted(e)
				If (Not Me.MyTableRowDeletedEvent Is Nothing) Then
					RaiseEvent MyTableRowDeleted(Me, New MyTableRowChangeEvent((CType(e.Row, MyTableRow)), e.Action))
				End If
			End Sub

			Protected Overrides Sub OnRowDeleting(ByVal e As DataRowChangeEventArgs)
				MyBase.OnRowDeleting(e)
				If (Not Me.MyTableRowDeletingEvent Is Nothing) Then
					RaiseEvent MyTableRowDeleting(Me, New MyTableRowChangeEvent((CType(e.Row, MyTableRow)), e.Action))
				End If
			End Sub

			Public Sub RemoveMyTableRow(ByVal row As MyTableRow)
				Me.Rows.Remove(row)
			End Sub
		End Class

		<System.Diagnostics.DebuggerStepThrough()> _
		Public Class MyTableRow
			Inherits DataRow

			Private tableMyTable As MyTableDataTable

			Friend Sub New(ByVal rb As DataRowBuilder)
				MyBase.New(rb)
				Me.tableMyTable = (CType(Me.Table, MyTableDataTable))
			End Sub

			Public Property id() As Integer
				Get
					Return (CInt(Fix(Me(Me.tableMyTable.idColumn))))
				End Get
				Set
					Me(Me.tableMyTable.idColumn) = Value
				End Set
			End Property

			Public Property colors() As String
				Get
					Try
						Return (CStr(Me(Me.tableMyTable.colorsColumn)))
					Catch e As InvalidCastException
						Throw New StrongTypingException("Cannot get value because it is DBNull.", e)
					End Try
				End Get
				Set
					Me(Me.tableMyTable.colorsColumn) = Value
				End Set
			End Property

			Public Property values() As Integer
				Get
					Try
						Return (CInt(Fix(Me(Me.tableMyTable.valuesColumn))))
					Catch e As InvalidCastException
						Throw New StrongTypingException("Cannot get value because it is DBNull.", e)
					End Try
				End Get
				Set
					Me(Me.tableMyTable.valuesColumn) = Value
				End Set
			End Property

			Public Function IscolorsNull() As Boolean
				Return Me.IsNull(Me.tableMyTable.colorsColumn)
			End Function

			Public Sub SetcolorsNull()
				Me(Me.tableMyTable.colorsColumn) = System.Convert.DBNull
			End Sub

			Public Function IsvaluesNull() As Boolean
				Return Me.IsNull(Me.tableMyTable.valuesColumn)
			End Function

			Public Sub SetvaluesNull()
				Me(Me.tableMyTable.valuesColumn) = System.Convert.DBNull
			End Sub
		End Class

		<System.Diagnostics.DebuggerStepThrough()> _
		Public Class MyTableRowChangeEvent
			Inherits EventArgs

			Private eventRow As MyTableRow

			Private eventAction As DataRowAction

			Public Sub New(ByVal tableRow As MyTableRow, ByVal dataRowAction As DataRowAction)
				Me.eventRow = tableRow
				Me.eventAction = dataRowAction
			End Sub

			Public ReadOnly Property Row() As MyTableRow
				Get
					Return Me.eventRow
				End Get
			End Property

			Public ReadOnly Property Action() As DataRowAction
				Get
					Return Me.eventAction
				End Get
			End Property
		End Class
	End Class
End Namespace

File: Examples/Chart/WebForm/CSharp/App_Code/DataManipulation/Binding/dsTable.cs
namespace Nevron.Examples.Chart.WebForm {
    using System;
    using System.Data;
    using System.Xml;
    using System.Runtime.Serialization;
    
    [Serializable()]
    [System.ComponentModel.DesignerCategoryAttribute("code")]
    [System.Diagnostics.DebuggerStepThrough()]
    [System.ComponentModel.ToolboxItem(true)]
    public class dsTable : DataSet {
        
        private MyTableDataTable tableMyTable;
        
        public dsTable() {
            this.InitClass();
            System.ComponentModel.CollectionChangeEventHandler schemaChangedHandler = new System.ComponentModel.CollectionChangeEventHandler(this.SchemaChanged);
            this.Tables.CollectionChanged += schemaChangedHandler;
            this.Relations.CollectionChanged += schemaChangedHandler;
        }
        
        protected dsTable(SerializationInfo info, StreamingContext context) {
            string strSchema = ((string)(info.GetValue("XmlSchema", typeof(string))));
            if ((strSchema != null)) {
                DataSet ds = new DataSet();
                ds.ReadXmlSchema(new XmlTextReader(new System.IO.StringReader(strSchema)));
                if ((ds.Tables["MyTable"] != null)) {
                    this.Tables.Add(new MyTableDataTable(ds.Tables["MyTable"]));
                }
                this.DataSetName = ds.DataSetName;
                this.Prefix = ds.Prefix;
                this.Namespace = ds.Namespace;
                this.Locale = ds.Locale;
                this.CaseSensitive = ds.CaseSensitive;
                this.EnforceConstraints = ds.EnforceConstraints;
                this.Merge(ds, false, System.Data.MissingSchemaAction.Add);
                this.InitVars();
            }
            else {
                this.InitClass();
            }
            this.GetSerializationData(info, context);
            System.ComponentModel.CollectionChangeEventHandler schemaChangedHandler = new System.ComponentModel.CollectionChangeEventHandler(this.SchemaChanged);
            this.Tables.CollectionChanged += schemaChangedHandler;
            this.Relations.CollectionChanged += schemaChangedHandler;
        }
        
        [System.ComponentModel.Browsable(false)]
        [System.ComponentModel.DesignerSerializationVisibilityAttribute(System.ComponentModel.DesignerSerializationVisibility.Content)]
        public MyTableDataTable MyTable {
            get {
                return this.tableMyTable;
            }
        }
        
        public override DataSet Clone() {
            dsTable cln = ((dsTable)(base.Clone()));
            cln.InitVars();
            return cln;
        }
        
        protected override bool ShouldSerializeTables() {
            return false;
        }
        
        protected override bool ShouldSerializeRelations() {
            return false;
        }
        
        protected override void ReadXmlSerializable(XmlReader reader) {
            this.Reset();
            DataSet ds = new DataSet();
            ds.ReadXml(reader);
            if ((ds.Tables["MyTable"] != null)) {
                this.Tables.Add(new MyTableDataTable(ds.Tables["MyTable"]));
            }
            this.DataSetName = ds.DataSetName;
            this.Prefix = ds.Prefix;
            this.Namespace = ds.Namespace;
            this.Locale = ds.Locale;
            this.CaseSensitive = ds.CaseSensitive;
            this.EnforceConstraints = ds.EnforceConstraints;
            this.Merge(ds, false, System.Data.MissingSchemaAction.Add);
            this.InitVars();
        }
        
        protected override System.Xml.Schema.XmlSchema GetSchemaSerializable() {
            System.IO.MemoryStream stream = new System.IO.MemoryStream();
            this.WriteXmlSchema(new XmlTextWriter(stream, null));
            stream.Position = 0;
            return System.Xml.Schema.XmlSchema.Read(new XmlTextReader(stream), null);
        }
        
        internal void InitVars() {
            this.tableMyTable = ((MyTableDataTable)(this.Tables["MyTable"]));
            if ((this.tableMyTable != null)) {
                this.tableMyTable.InitVars();
            }
        }
        
        private void InitClass() {
            this.DataSetName = "dsTable";
            this.Prefix = "";
            this.Namespace = "http://www.tempuri.org/dsTable.xsd";
            this.Locale = new System.Globalization.CultureInfo("bg-BG");
            this.CaseSensitive = false;
            this.EnforceConstraints = true;
            this.tableMyTable = new MyTableDataTable();
            this.Tables.Add(this.tableMyTable);
        }
        
        private bool ShouldSerializeMyTable() {
            return false;
        }
        
        private void SchemaChanged(object sender, System.ComponentModel.CollectionChangeEventArgs e) {
            if ((e.Action == System.ComponentModel.CollectionChangeAction.Remove)) {
                this.InitVars();
            }
        }
        
        public delegate void MyTableRowChangeEventHandler(object sender, MyTableRowChangeEvent e);
        
        [System.Diagnostics.DebuggerStepThrough()]
        public class MyTableDataTable : DataTable, System.Collections.IEnumerable {
            
            public DataColumn columnid;
            public DataColumn columncolors;
            public DataColumn columnvalues;
            
            internal MyTableDataTable() : 
                    base("MyTable") {
                this.InitClass();
            }
            
            internal MyTableDataTable(DataTable table) : 
                    base(table.TableName) {
                if ((table.CaseSensitive != table.DataSet.CaseSensitive)) {
                    this.CaseSensitive = table.CaseSensitive;
                }
                if ((table.Locale.ToString() != table.DataSet.Locale.ToString())) {
                    this.Locale = table.Locale;
                }
                if ((table.Namespace != table.DataSet.Namespace)) {
                    this.Namespace = table.Namespace;
                }
                this.Prefix = table.Prefix;
                this.MinimumCapacity = table.MinimumCapacity;
                this.DisplayExpression = table.DisplayExpression;
            }
            
            [System.ComponentModel.Browsable(false)]
            public int Count {
                get {
                    return this.Rows.Count;
                }
            }
            
            internal DataColumn idColumn {
                get {
                    return this.columnid;
                }
            }
            
            internal DataColumn colorsColumn {
                get {
                    return this.columncolors;
                }
            }
            
            internal DataColumn valuesColumn {
                get {
                    return this.columnvalues;
                }
            }
            
            public MyTableRow this[int index] {
                get {
                    return ((MyTableRow)(this.Rows[index]));
                }
            }
            
            public event MyTableRowChangeEventHandler MyTableRowChanged;
            
            public event MyTableRowChangeEventHandler MyTableRowChanging;
            
            public event MyTableRowChangeEventHandler MyTableRowDeleted;
            
            public event MyTableRowChangeEventHandler MyTableRowDeleting;
            
            public void AddMyTableRow(MyTableRow row) {
                this.Rows.Add(row);
            }
            
            public MyTableRow AddMyTableRow(string colors, int values) {
                MyTableRow rowMyTableRow = ((MyTableRow)(this.NewRow()));
                rowMyTableRow.ItemArray = new object[] {
                        null,
                        colors,
                        values};
                this.Rows.Add(rowMyTableRow);
                return rowMyTableRow;
            }
            
            public MyTableRow FindByid(int id) {
                return ((MyTableRow)(this.Rows.Find(new object[] {
                            id})));
            }
            
            public System.Collections.IEnumerator GetEnumerator() {
                return this.Rows.GetEnumerator();
            }
            
            public override DataTable Clone() {
                MyTableDataTable cln = ((MyTableDataTable)(base.Clone()));
                cln.InitVars();
                return cln;
            }
            
            protected override DataTable CreateInstance() {
                return new MyTableDataTable();
            }
            
            internal void InitVars() {
                this.columnid = this.Columns["id"];
                this.columncolors = this.Columns["colors"];
                this.columnvalues = this.Columns["values"];
            }
            
            private void InitClass() {
                this.columnid = new DataColumn("id", typeof(int), null, System.Data.MappingType.Element);
                this.Columns.Add(this.columnid);
                this.columncolors = new DataColumn("colors", typeof(string), null, System.Data.MappingType.Element);
                this.Columns.Add(this.columncolors);
                this.columnvalues = new DataColumn("values", typeof(int), null, System.Data.MappingType.Element);
                this.Columns.Add(this.columnvalues);
                this.Constraints.Add(new UniqueConstraint("Constraint1", new DataColumn[] {
                                this.columnid}, true));
                this.columnid.AutoIncrement = true;
                this.columnid.AllowDBNull = false;
                this.columnid.Unique = true;
            }
            
            public MyTableRow NewMyTableRow() {
                return ((MyTableRow)(this.NewRow()));
            }
            
            protected override DataRow NewRowFromBuilder(DataRowBuilder builder) {
                return new MyTableRow(builder);
            }
            
            protected override System.Type GetRowType() {
                return typeof(MyTableRow);
            }
            
            protected override void OnRowChanged(DataRowChangeEventArgs e) {
                base.OnRowChanged(e);
                if ((this.MyTableRowChanged != null)) {
                    this.MyTableRowChanged(this, new MyTableRowChangeEvent(((MyTableRow)(e.Row)), e.Action));
                }
            }
            
            protected override void OnRowChanging(DataRowChangeEventArgs e) {
                base.OnRowChanging(e);
                if ((this.MyTableRowChanging != null)) {
                    this.MyTableRowChanging(this, new MyTableRowChangeEvent(((MyTableRow)(e.Row)), e.Action));
                }
            }
            
            protected override void OnRowDeleted(DataRowChangeEventArgs e) {
                base.OnRowDeleted(e);
                if ((this.MyTableRowDeleted != null)) {
                    this.MyTableRowDeleted(this, new MyTableRowChangeEvent(((MyTableRow)(e.Row)), e.Action));
                }
            }
            
            protected override void OnRowDeleting(DataRowChangeEventArgs e) {
                base.OnRowDeleting(e);
                if ((this.MyTableRowDeleting != null)) {
                    this.MyTableRowDeleting(this, new MyTableRowChangeEvent(((MyTableRow)(e.Row)), e.Action));
                }
            }
            
            public void RemoveMyTableRow(MyTableRow row) {
                this.Rows.Remove(row);
            }
        }
        
        [System.Diagnostics.DebuggerStepThrough()]
        public class MyTableRow : DataRow {
            
            private MyTableDataTable tableMyTable;
            
            internal MyTableRow(DataRowBuilder rb) : 
                    base(rb) {
                this.tableMyTable = ((MyTableDataTable)(this.Table));
            }
            
            public int id {
                get {
                    return ((int)(this[this.tableMyTable.idColumn]));
                }
                set {
                    this[this.tableMyTable.idColumn] = value;
                }
            }
            
            public string colors {
                get {
                    try {
                        return ((string)(this[this.tableMyTable.colorsColumn]));
                    }
                    catch (InvalidCastException e) {
                        throw new StrongTypingException("Cannot get value because it is DBNull.", e);
                    }
                }
                set {
                    this[this.tableMyTable.colorsColumn] = value;
                }
            }
            
            public int values {
                get {
                    try {
                        return ((int)(this[this.tableMyTable.valuesColumn]));
                    }
                    catch (InvalidCastException e) {
                        throw new StrongTypingException("Cannot get value because it is DBNull.", e);
                    }
                }
                set {
                    this[this.tableMyTable.valuesColumn] = value;
                }
            }
            
            public bool IscolorsNull() {
                return this.IsNull(this.tableMyTable.colorsColumn);
            }
            
            public void SetcolorsNull() {
                this[this.tableMyTable.colorsColumn] = System.Convert.DBNull;
            }
            
            public bool IsvaluesNull() {
                return this.IsNull(this.tableMyTable.valuesColumn);
            }
            
            public void SetvaluesNull() {
                this[this.tableMyTable.valuesColumn] = System.Convert.DBNull;
            }
        }
        
        [System.Diagnostics.DebuggerStepThrough()]
        public class MyTableRowChangeEvent : EventArgs {
            
            private MyTableRow eventRow;
            
            private DataRowAction eventAction;
            
            public MyTableRowChangeEvent(MyTableRow tableRow, DataRowAction dataRowAction) {
                this.eventRow = tableRow;
                this.eventAction = dataRowAction;
            }
            
            public MyTableRow Row {
                get {
                    return this.eventRow;
                }
            }
            
            public DataRowAction Action {
                get {
                    return this.eventAction;
                }
            }
        }
    }
}
